products/bigquery/api.yaml

```diff
@@ -1433,3 +1433,139 @@ objects:
                         - :INTEGER
                         - :FLOAT
                         - :BOOLEAN
+  - !ruby/object:Api::Resource
+    name: 'Routine'
+    kind: 'bigquery#routine'
+    base_url: projects/{{project}}/datasets/{{dataset_id}}/routines
+    self_link: projects/{{project}}/datasets/{{dataset_id}}/routines/{{routine_id}}
+    description: |
+      A user-defined function or a stored procedure that belongs to a Dataset
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Routines Intro': 'https://cloud.google.com/bigquery/docs/reference/rest/v2/routines'
+      api: 'https://cloud.google.com/bigquery/docs/reference/rest/v2/routines'
+    properties:
+      - !ruby/object:Api::Type::NestedObject
+        name: routineReference
+        description: Reference describing the ID of this routine
+        required: true
+        properties:
+          - !ruby/object:Api::Type::String
+            name: 'datasetId'
+            description: The ID of the dataset containing this routine
+            required: true
+          - !ruby/object:Api::Type::String
+            name: 'projectId'
+            description: The ID of the project containing this routine
+          - !ruby/object:Api::Type::String
+            name: 'routineId'
+            description: The ID of the the routine. The ID must contain only
+              letters (a-z, A-Z), numbers (0-9), or underscores (_).
+              The maximum length is 256 characters.
+            required: true
+      - !ruby/object:Api::Type::Enum
+        name: 'routineType'
+        description: The type of routine.
+        values:
+          - :ROUTINE_TYPE_UNSPECIFIED
```

Comment:
```suggestion
```

`_UNSPECIFIED` enum values are usually an artifact of some internal rules that protobuf enums need to have their first value be `_UNSPECIFIED` or some kind of default. They generally aren't configurable, so we can probably remove this. If you've found it is configurable, feel free to ignore this!

Reply:
removed